File: 0896-monotonic-array/0896-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& a) {
     int n=a.size();
     bool inc=true;
     bool dec=true;
     for(int i=0;i<n-1;i++){
         if(a[i]>a[i+1]) inc=false;
         if(a[i]<a[i+1]) dec=false;
         if(inc == false and dec == false) return false;
     }
        return true;
    }
};

File: 0896-monotonic-array/0896-monotonic-array.rs
impl Solution {
    pub fn is_monotonic(nums: Vec<i32>) -> bool {
        let mut inc:bool = true;
        let mut dec:bool = true;
        let n = nums.len();
        for i in 0..n-1{
            if nums[i]>nums[i+1] {inc = false;}
            if nums[i]<nums[i+1] {dec = false;}
            if !inc && !dec{
                return false;
            } 
        }
        true
    }
}
